src/base/CodeContext.h

```diff
@@ -14,6 +14,41 @@
 
 #include <iosfwd>
 
+/** \file
+ *
+ * Most error-reporting code cannot know what transaction or task Squid was
+ * working on when the error occurred. For example, when Squid HTTP request
+ * parser discovers a malformed header field, the parser can report the field
+ * contents, but that information is often useless for the admin without
+ * processing context details like which client sent the request or what the
+ * requested URL was. Moreover, even when the error reporting code does have
+ * access to some context details, it cannot separate important facts from noise
+ * because such classification is usually deployment-specific (i.e. cannot be
+ * hard-coded) and requires human expertise. The situation is aggravated by a
+ * busy Squid instance constantly switching from one processing context to
+ * another.
+ *
+ * To solve these problems, Squid maintains a CodeContext object tied to
+ * (eventually) each processing context. CodeContext objects can report context
+ * details that can be tied to access.log records. When Squid switches to
+ * another processing context, it switches the current CodeContext object as
+ * well. When Squid prints a level-0 or level-1 message to cache.log, it asks
+ * the current CodeContext object (if any) to report context details, allowing
+ * the admin to correlate the error message with an access.log record. In most
+ * cases, the processing context is a master transaction that Squid is working
+ * on, which is reported using a master transaction ID and can be recorded in
+ * access.log using a %master_xaction logformat code.
+ *
+ * Squid also reports processing context changes to cache.log when Squid
+ * level-5+ debugging is enabled.
+ *
+ * CodeContext is being retrofitted into existing code with lots of places that
+ * switch processing context. Identifying and adjusting all those places takes
+ * time. Until then, there will be incorrect and missing context attributions.
+ *
```

Comment:
> Suggested update to class CodeContext description: ...

I had too many reservations about that description to use it, but I hope that this text added in branch commit 880a53d is enough to move this (unrelated to CodeContext documentation!) PR forward.

Reply:
This one could be less verbose, but I can accept different wording.